circus-train-core/src/test/java/com/hotels/bdp/circustrain/core/metastore/MetaStoreClientFactoryManagerTest.java

```diff
@@ -20,25 +20,27 @@
 import java.util.Collections;
 import java.util.List;
 
+import org.apache.hadoop.hive.conf.HiveConf;
 import org.junit.Test;
 
-import com.hotels.bdp.circustrain.api.CircusTrainException;
-import com.hotels.bdp.circustrain.api.metastore.MetaStoreClientFactory;
+import com.hotels.hcommon.hive.metastore.client.api.MetaStoreClientFactory;
 
 public class MetaStoreClientFactoryManagerTest {
 
   @Test
   public void factoryForThrift() {
-    List<MetaStoreClientFactory> list = Collections
-        .<MetaStoreClientFactory> singletonList(new ThriftMetaStoreClientFactory());
+    List<ConditionalMetaStoreClientFactory> list = Collections
+        .<ConditionalMetaStoreClientFactory> singletonList(
+            new ThriftHiveMetaStoreClientFactory(new HiveConf(), "name"));
     MetaStoreClientFactoryManager factoryManager = new MetaStoreClientFactoryManager(list);
-    MetaStoreClientFactory clientFactory = factoryManager.factoryForUrl(ThriftMetaStoreClientFactory.ACCEPT_PREFIX);
-    assertTrue(clientFactory instanceof ThriftMetaStoreClientFactory);
+    MetaStoreClientFactory clientFactory = factoryManager.factoryForUrl(
+        ThriftHiveMetaStoreClientFactory.ACCEPT_PREFIX);
+    assertTrue(clientFactory instanceof ThriftHiveMetaStoreClientFactory);
   }
 
-  @Test(expected = CircusTrainException.class)
+  @Test(expected = RuntimeException.class)
```

Comment:
expecting `Runtime` is very generic, usually not a good idea as it can come from anywhere. Can you make it more specific or make sure it gets throw on the exact call that you want

Reply:
Changed